File: VB/DXWebApplication5/SalesPersonData.vb
Imports System
Imports System.Collections.Generic

Namespace DXWebApplication5
	Public Class SalesPersonData
		Public Property SalesPerson() As String
		Public Property Quantity() As Integer
		Public Property Checked() As Boolean
	End Class

	Public NotInheritable Class DataGenerator

		Private Sub New()
		End Sub
		Private Shared _data As List(Of SalesPersonData)

		Public Shared ReadOnly Property Data() As List(Of SalesPersonData)
			Get
				If _data Is Nothing Then
					_data = CreateSourceData()
				End If
				Return _data
			End Get
		End Property
		Public Shared Function CreateSourceData() As List(Of SalesPersonData)
'INSTANT VB NOTE: The variable data was renamed since Visual Basic does not handle local variables named the same as class members well:
			Dim data_Renamed As New List(Of SalesPersonData)()
			Dim salesPersons() As String = { "Andrew Fuller", "Michael Suyama", "Robert King", "Nancy Davolio", "Margaret Peacock", "Laura Callahan", "Steven Buchanan", "Janet Leverling" }
			Dim seed As Integer = CInt(Date.Now.Ticks) And &HFFFF
			Dim rand As New Random(seed)

			For i As Integer = 0 To 9999
				Dim record As New SalesPersonData()

				record.SalesPerson = salesPersons(rand.Next(0, salesPersons.Length))
				record.Quantity = (New Random(seed)).Next(0, 100)
				data_Renamed.Add(record)
			Next i
			Return data_Renamed
		End Function
	End Class
End Namespace

File: CS/DXWebApplication5/SalesPersonData.cs
using System;
using System.Collections.Generic;

namespace DXWebApplication5
{
    public class SalesPersonData {
        public string SalesPerson { get; set; }
        public int Quantity { get; set; }
        public bool Checked { get; set; }
    }

    public static class DataGenerator {
        private static List<SalesPersonData> _data;

        public static List<SalesPersonData> Data
        {
            get {
                if (_data == null) {
                    _data = CreateSourceData();
                }
                return _data;
            }
        }
        public static List<SalesPersonData> CreateSourceData() {
            List<SalesPersonData> data = new List<SalesPersonData>();
            string[] salesPersons = { "Andrew Fuller", "Michael Suyama",
                                    "Robert King", "Nancy Davolio",
                                    "Margaret Peacock", "Laura Callahan",
                                    "Steven Buchanan", "Janet Leverling" };
            int seed = (int)DateTime.Now.Ticks & 0x0000FFFF;
            Random rand = new Random(seed);

            for (int i = 0; i < 10000; i++) {
                SalesPersonData record = new SalesPersonData();

                record.SalesPerson = salesPersons[rand.Next(0, salesPersons.Length)];
                record.Quantity = new Random(seed).Next(0, 100);
                data.Add(record);
            }
            return data;
        }
    }
}